src/commands/chatInput/data/reset/scoreboard.ts

```diff
@@ -0,0 +1,59 @@
+import { ButtonStyle, ComponentType } from "discord.js";
+import type { ChatInputCommand } from "../..";
+import { components } from "../../../../handlers/interactions/components";
+import { databaseLogger } from "../../../../utils/logger/database";
+
+const command: ChatInputCommand = {
+  description: "Reset the guild scoreboard",
+  execute(interaction, _, document, [countingChannelId, countingChannel]) {
+    void interaction.reply({
+      content: `⚠ Are you sure you want to reset the scoreboard of <#${countingChannelId!}>? This action cannot be undone.`,
+      components: [
+        {
+          type: ComponentType.ActionRow,
+          components: [
+            {
+              type: ComponentType.Button,
+              label: "No, go back",
+              customId: `${interaction.id}:no`,
+              style: ButtonStyle.Secondary,
+            },
+            {
+              type: ComponentType.Button,
+              label: "Yes, I'm sure",
+              customId: `${interaction.id}:yes`,
+              style: ButtonStyle.Danger,
+            },
+          ],
+        },
+      ],
+    });
+
+    components.set(`${interaction.id}:yes`, {
+      type: "BUTTON",
+      allowedUsers: [interaction.user.id],
+      callback: button => {
+        databaseLogger.verbose(`Cleared scoreboard of guild ${interaction.guildId}, contents were ${JSON.stringify(countingChannel?.scores)}`);
```

Comment:
how will this look when`.scores` is a map?

Reply:
As a normal object.

```log
2022-11-11 03:25:41 verbose: 	Cleared scoreboard of guild 781780421103976449, contents were {"462870395314241537":23}
```

Reply:
just checked, looks like it's buggy in some cases. idk if mongoose handles that somehow though.
![image](https://user-images.githubusercontent.com/10573728/201315803-6d8081a8-780e-4131-aa1e-e9510e7cb7ea.png)

I mean if it works then it works 😅